Add FileTree component to docs (#12623)
flutter/website: src/content/app-architecture/case-study/index.md

while UI layer objects (views and view models) are.
The following is how the code is organized within the Compass application.

<FileTree>

- lib/
  - ui/
    - core/
      - ui/ 
        - <shared_widgets>
      - themes/
    - <feature_name>/
      - view_model/
        - <view_model_class>.dart
      - widgets/
        - <feature_name>_screen.dart
        - <other_widgets>
  - domain/
    - models/
      - <model_name>.dart
  - data/
    - repositories/
      - <repository_class>.dart
    - services/
      - <service_class>.dart
    - model/
      - <api_model_class>.dart
  - config/
  - utils/
  - routing/
  - main_staging.dart
  - main_development.dart
  - main.dart

> - test/ // The test folder contains unit and widget tests

Reviewer: Since the comments are already styled differently, can we drop the `//` or do you think they're still helpful visually?

  - data/
  - domain/
  - ui/
  - utils/

> - testing/ // The testing folder contains mocks other classes need to execute tests

Reviewer: ```suggestion
- testing/ // Contains mocks that other classes need to execute tests.
```

Author: Done
